website/scripts/send-data-to-vanta.js

```diff
@@ -0,0 +1,288 @@
+          authMethod = 'PASSWORD';
+        }
+
+        // Set the permissionLevel using the user's global_role value.
+        let permissionLevel = 'BASE';
+        if(user.global_role === 'admin'){
+          permissionLevel = 'ADMIN';
+        } else if(user.global_role === 'maintainer'){
+          permissionLevel = 'EDITOR';
+        }
+
+        // Create a user object for this Fleet user.
+        let userToSyncWithVanta = {
+          displayName: user.name,
+          uniqueId: String(user.id),// Vanta requires this value to be a string.
+          fullName: user.name,
+          accountName: user.name,
+          email: user.email,
+          createdTimestamp: user.created_at,
+          status: 'ACTIVE',// Always set to 'ACTIVE', if a user is removed from the Fleet instance, it will not be included in the response from the Fleet instance's /users endpoint.
+          mfaEnabled: false,
+          mfaMethods: ['DISABLED'],
+          externalUrl: vantaConnection.fleetInstanceUrl,// Setting externalUrl (Required by Vanta) for all users to be the Fleet instance url.
+          authMethod,
+          permissionLevel,
+        };
+
+        // Add the user object to the array of users to sync with Vanta.
+        usersToSyncWithVanta.push(userToSyncWithVanta);
+      }
+
+
+      //  ┌─┐┌─┐┌┬┐  ┬ ┬┌─┐┌─┐┌┬┐┌─┐
+      //  │ ┬├┤  │   ├─┤│ │└─┐ │ └─┐
+      //  └─┘└─┘ ┴   ┴ ┴└─┘└─┘ ┴ └─┘
+      // Get all hosts on the Fleet instance.
+      let pageNumberForPossiblePaginatedResults = 0;
+      let numberOfHostsPerRequest = 100;
+      let allHostsOnThisFleetInstance = [];
+
+      // Start sending requests to the Fleet instance's /hosts endpoint until we have all hosts.
+      await sails.helpers.flow.until(async ()=>{
+        let getHostsResponse = await sails.helpers.http.get(
+          `${updatedRecord.fleetInstanceUrl}/api/v1/fleet/hosts?per_page=${numberOfHostsPerRequest}&page=${pageNumberForPossiblePaginatedResults}`,
+          {},
+          {'Authorization': 'bearer '+updatedRecord.fleetApiKey},
+        );
+        // Add the results to the allHostsOnThisFleetInstance array.
+        allHostsOnThisFleetInstance = allHostsOnThisFleetInstance.concat(getHostsResponse.hosts);
+        // Increment the page of results we're requesting.
+        pageNumberForPossiblePaginatedResults++;
+        // If we recieved less results than we requested, we've reached the last page of the results.
+        return getHostsResponse.hosts.length !== numberOfHostsPerRequest;
+      }, 10000).intercept('tookTooLong', ()=>{// If an error occurs while sending a request to the Fleet instance, we'll add the error to the errorReportById object, with this connections ID set as the key.
+        errorReportById[connectionIdAsString] = new Error(`When requesting all hosts from a Fleet instance for a VantaConnection (id: ${connectionIdAsString}), the Fleet instance did not respond with all of it's hosts in the set amount of time.`);
+      });
```

Comment:
use `.tolerate()` instead of intercept here.  Keep in mind that the return value of the tolerate callback becomes the return value of the entire thing

Also I'd put the `.tolerate` on a new line because it can be harder to read (e.g. the reason I starrted talking about return values in my comment just now when actually I dont think it matters